build_tools/benchmarks/comparisons/common/benchmark_runner.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import re
+import subprocess
+import time
+
+from common.benchmark_command import BenchmarkCommand
+
+# Regexes for retrieving memory information.
+_VMHWM_REGEX = re.compile(r".*?VmHWM:.*?(\d+) kB.*")
+_VMRSS_REGEX = re.compile(r".*?VmRSS:.*?(\d+) kB.*")
+_RSSFILE_REGEX = re.compile(r".*?RssFile:.*?(\d+) kB.*")
+
+
+def run_command(benchmark_command: BenchmarkCommand) -> list[float]:
+  """Runs `benchmark_command` and polls for memory consumption statistics.
+  Args:
+    benchmark_command: A `BenchmarkCommand` object containing information on how to run the benchmark and parse the output.
+  Returns:
+    An array containing values for [`latency`, `vmhwm`, `vmrss`, `rssfile`]
+  """
+  command = benchmark_command.generate_benchmark_command()
+  print("\n\nRunning command:\n" + " ".join(command))
+  benchmark_process = subprocess.Popen(command, stdout=subprocess.PIPE,
+                                       stderr=subprocess.STDOUT)
+
+  # Keep a record of the highest VmHWM corresponding VmRSS and RssFile values.
```

Comment:
This isn't really guaranteed to actually capture the other values corresponding to the highest VmHWM. Only whatever they were within half a second of the peak, right?

Reply:
You're right. We might rework this to record the median values instead of VmHWM, given the memory usage typically looks more like this:

![image](https://user-images.githubusercontent.com/5143063/170343039-5c7dcda5-5e48-4d84-b9ec-822d87e902ea.png)

We might be more interested in vmrss instead of vmhwm.